hipoexceptionsandroid/src/test/java/com/hipo/hipoexceptionsandroid/SampleDataUnitTest.kt

```diff
@@ -0,0 +1,244 @@
+package com.hipo.hipoexceptionsandroid
+
+import com.google.gson.Gson
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Call
+import java.io.*
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class SampleDataUnitTest {
+
+    interface TestService {
+        @GET("/")
+        fun getString(): Call<String>
+
+        @POST("/")
+        fun postString(@Body requestString: String): Call<String>
+    }
+
+    private var server = MockWebServer()
+    private lateinit var testService: TestService
+
+    private val defaultErrorMessage = "New Generic - An Error Occured"
+
+    private val retrofitExceptionHandler =
+        RetrofitErrorHandler(
+            Gson(),
+            defaultErrorMessage,
+            intArrayOf(402)
+        )
+
+    @Before
+    fun setup() {
```

Comment:
JUnit4'te her test methodu için bir sınıf oluşturuluyor. Bu sebeple setup'ın static olması gerek. Kotlinde static olmadığı için JUnit5 ile yeni bir annotation geldi. @TestInstance(Lifecycle.PER_CLASS) annotation'ınını sınıfın başına eklediğin zaman, her method için tek bir sınıf instance'ı kullanılıyor. Belki güncellemek isteyebilirsin. 

https://medium.com/@m.sinan.sari/junit5-kotlin-7f937a6aef35

Reply:
I'll switch to Junit5 later then. Thanks for letting me know.